### DISCUSSION TITLE
Custom shortcodes to avoid hardcoding journal citation

### DISCUSSION THREAD

**Discussion by materiajournal** [thegetty/quire#131] [Get Help] [answered]
I’m building an about page for our journal, and would like to include citation information for the journal as a whole avoiding hardcoding. ls it possible to create a shortcode to pull that information from the metadata? How difficult would it be to do this?

I was hoping to implement something similar to the side-menu, which is a partial that pulls the citation info for each page from metadata files. Obviously, I cannot use partials with markdown pages, but am yet to figure out how I might do this with a shortcode.

I have read the Hugo docs on shortcodes and partials, and searched GitHub, but hoped somebody might be able to point me in the right direction.

Thanks,
Sophie

**Comment by geealbers** [comment, marked as answer]
Hi Sophie, when you said:

> I cannot use partials with markdown pages, but am yet to figure out how I might do this with a shortcode.

You're absolutely correct that partials can't be used directly in Markdown pages, but the nice thing is that you can usually create a very simple shortcode that acts as a wrapper for an existing partial.

If you had a shortcode called `q-cite-this.thml` with this code inside it:

```
{{- $type := .Get "type" -}}
{{- $range := .Get "range" -}}

<p>{{ partial "cite-this.html" ( dict "page" . "site" .Site "type" $type "range" $range ) }}</p>
```

You could then call it in the markdown, as:

```md
## Citation Information

### Chicago

{{< q-cite-this type="chicago" range="site" >}}

### MLA

{{< q-cite-this type="mla" range="site" >}}
```

(`type` can be "chicago" or "mla", and `range` can be "page" or "site")

That said, a **HUGE CAVEAT**,  the cite-this partial is actually not all that robust and has not at this point been written with journals like *Materia* in mind. So, I think you're going to find that the resulting output is pretty wonky. Try it out though, and if you get the shortcode working, I'd be happy to work on the cite-this partial to see if we can fix the output. I'm sure it would be useful for other community members as well. I'd just need exact samples of what both the Chicago and MLA citations should ideally look like in your case.

**Reply by Erin-Cecele** [reply]
Hi @materiajournal and @sophielamb. Did this help answer your question or are you in need of further assistance?

**Reply by sophielamb** [reply]
Hi @Erin-Cecele, thanks so much for getting in touch. Regarding this thread - I did make a shortcode, but we didn't use it in the end as we decided the existing citations were suitable. In the end I dipped into the _contents-list.html_ partial to make minor changes instead of creating a separate "about" page.

At the moment however, I am still working on writing custom shortcodes for alternative ways to display images (a before/after figure with draggable slider - two superimposed images with a CSS 'button' and JQuery to make it work). I have the vanilla HTML/CSS/JQuery and used an existing figure shortcode as a template to drop the components into. It does function, but needs work and will let you know how that goes.

We're in the early stages of the next issue, so will likely be in touch once we have a better idea what we want to achieve with it!

**Reply by Erin-Cecele** [reply]
Excellent @sophielamb. I look forward to hearing from you and seeing how the next issue shapes up!
